libs/dh/metering-point/routing/src/lib/dh-metering-point-overview.guard.spec.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2021 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Location } from '@angular/common';
+import { Component } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { Router } from '@angular/router';
+import { SpectacularAppComponent } from '@ngworker/spectacular';
+import { render, RenderResult } from '@testing-library/angular';
+
+import { DhMeteringPointOverviewGuard } from './dh-metering-point-overview.guard';
+import { dhMeteringPointIdParam } from './dh-metering-point-id-param';
+import { dhMeteringPointPath } from './dh-metering-point-path';
+
+describe(DhMeteringPointOverviewGuard.name, () => {
+  @Component({
+    template: '',
+  })
+  class TestGuardedComponent {}
+
+  @Component({
+    template: '',
+  })
+  class TestMeteringPointSearchComponent {}
+
+  beforeEach(async () => {
+    view = await render(SpectacularAppComponent, {
+      declarations: [TestGuardedComponent, TestMeteringPointSearchComponent],
+      routes: [
+        {
+          path: dhMeteringPointPath,
+          children: [
+            {
+              component: TestMeteringPointSearchComponent,
+              path: 'search',
+            },
+            {
+              canActivate: [DhMeteringPointOverviewGuard],
+              component: TestGuardedComponent,
+              path: `:${dhMeteringPointIdParam}`,
+            },
+          ],
+        },
+      ],
+    });
+    angularLocation = TestBed.inject(Location);
+    router = TestBed.inject(Router);
+  });
+
+  let angularLocation: Location;
+  let router: Router;
+  const validMeteringPointId = '123456789000000000';
```

Comment:
Would be nice to have valid / invalid metering point id's in some test-util. I can see multiple places where these would be useful :)

Reply:
I was thinking to extracting the metering point id to a library as well but thought that it might become too distant. Anyway, I just pushed commits that does that. You can see how it looks/works.